### PR TITLE
testing desktop tests in gha

### PR DESCRIPTION
**Pull request by ondracja**
- adding desktop PW tests to github actions
- disabling desktop PW tests in gitlab flow (will remove it completely after some "prod" testing)

Things that need to be solved in follow-up PRs:
- make tests successfully use `node-bridge` instead of `legacy-bridge`
- adding manual screenshots to artefacts (they are added locally but not in CI)
- figure out a way to include automated screenshots into desktop PW similar to web one (check [here](https://github.com/microsoft/playwright/issues/8208))

### CODE REVIEW

File: packages/suite-desktop-core/e2e/support/pageActions/dashboardActions.ts [hunk 1]
```diff
@@ -7,8 +7,23 @@ import { waitForDataTestSelector } from '../common';
 class DashboardActions {
     async passThroughInitialRun(window: Page) {
         await waitForDataTestSelector(window, '@welcome/title');
+        await window.screenshot({
+            path: './test-results/zacatek-initial-runu.png',
```
Comment (mroz22) [inline packages/suite-desktop-core/e2e/support/pageActions/dashboardActions.ts]: .

File: packages/suite-desktop-core/e2e/support/common.ts [hunk 2]
```diff
@@ -4,17 +4,20 @@ import { Page, _electron as electron } from '@playwright/test';
...
     const electronApp = await electron.launch({
         cwd: appDir,
         args: [
             path.join(appDir, './dist/app.js'),
             `--log-level=${desiredLogLevel}`,
-            '--bridge-node-test',
+            // '--bridge-node-test',
             // uncomment to use legacy bridge
-            // '--bridge-legacy-test'
+            '--bridge-legacy-test',
```
Comment (mroz22) [inline packages/suite-desktop-core/e2e/support/common.ts:21]: not testing production bridge anymore? :(

File: ci/test.yml [hunk 3]
```diff
@@ -6,55 +6,55 @@
...
-suite desktop manual:
-  extends: .e2e desktop
-  except:
-    <<: *run_everything_rules
-  when: manual
+# suite desktop manual:
+#   extends: .e2e desktop
+#   except:
+#     <<: *run_everything_rules
+#   when: manual
```
Comment (mroz22) [inline ci/test.yml]: rm?

File: .github/workflows/test-suite-desktop-e2e.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        include:
+          - TEST_FILE: "spawn-bridge"
+            CONTAINERS: "trezor-user-env-unix"
+          - TEST_FILE: "suite-guide"
+            CONTAINERS: "trezor-user-env-unix"
+          - TEST_FILE: "wallet-discovery"
+            CONTAINERS: "trezor-user-env-unix"
+          # - TEST_FILE: "cardano-discovery"
+          #   CONTAINERS: "trezor-user-env-unix"
+          # - TEST_FILE: "eap-modal"
```
Comment (mroz22) [inline .github/workflows/test-suite-desktop-e2e.yml:35]: why changes in eap model but here commented out?

File: .github/workflows/test-suite-desktop-e2e.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+      fail-fast: false
+      matrix:
+        include:
+          - TEST_FILE: "spawn-bridge"
+            CONTAINERS: "trezor-user-env-unix"
+          - TEST_FILE: "suite-guide"
+            CONTAINERS: "trezor-user-env-unix"
+          - TEST_FILE: "wallet-discovery"
+            CONTAINERS: "trezor-user-env-unix"
+          # - TEST_FILE: "cardano-discovery"
```
Comment (mroz22) [inline .github/workflows/test-suite-desktop-e2e.yml:33]: unrelated but what is the difference between wallet-discovery and cardano-discovery?

File: .github/workflows/test-suite-desktop-e2e.yml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        with:
+          node-version-file: ".nvmrc"
+          cache: yarn
+
+      - name: Install deps and build libs
+        run: |
+          yarn install --immutable
+          yarn message-system-sign-config
+          yarn workspace @trezor/suite-data build:lib
+          yarn workspace @trezor/connect-iframe build:lib
```
Comment (mroz22) [inline .github/workflows/test-suite-desktop-e2e.yml]: is it still needed? I remember @komret was trying to get rid of the need for connect-iframe build?

File: .github/workflows/test-suite-desktop-e2e.yml [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+          yarn workspace @trezor/transport-bridge build:lib
+
+      - name: Build linux suite-desktop
+        run: |
+          yarn workspace @trezor/suite-desktop build:linux
+          bash packages/suite-desktop-core/scripts/gnupg-sign.sh
```
Comment (mroz22) [inline .github/workflows/test-suite-desktop-e2e.yml]: is such a complicated build needed? doesn't it necessarily eat much CPU time? just app.js should be enough, isn't it?

File: packages/suite-desktop-core/e2e/support/pageActions/dashboardActions.ts (hunk 1 above)
Comment (ondracja) [reply packages/suite-desktop-core/e2e/support/pageActions/dashboardActions.ts]: Pw for electro has a problem with automated screenshot saving so I'm experimenting with manual screens until it's solved (I have a task for this for a later point in time).

On the second thought, I will remove this and later add it in a more thought out form 😄

File: ci/test.yml (hunk 3 above)
Comment (ondracja) [reply ci/test.yml]: removed.

File: .github/workflows/test-suite-desktop-e2e.yml (hunk 5 above)
Comment (ondracja) [reply .github/workflows/test-suite-desktop-e2e.yml:33]: `wallet-discovery` checks only standard btc acc discovery, `cardano-discovery` checks that discovery finishes with all ada acc types enabled. I have a test case refactor in mind for this as well.

File: .github/workflows/test-suite-desktop-e2e.yml (hunk 7 above)
Comment (ondracja) [reply .github/workflows/test-suite-desktop-e2e.yml]: Changed to a simplified build.

File: packages/suite-desktop-core/e2e/support/common.ts (hunk 2 above)
Comment (ondracja) [reply packages/suite-desktop-core/e2e/support/common.ts:21]: switched back to node-bridge, will keep legacy bridge for local testing only

File: packages/suite-desktop-core/e2e/support/common.ts [hunk 8]
```diff
@@ -4,17 +4,20 @@ import { Page, _electron as electron } from '@playwright/test';
 import path from 'path';
 import fs from 'fs';
 
+import { TrezorUserEnvLink } from '@trezor/trezor-user-env-link';
+
 export const launchSuite = async () => {
     const appDir = path.join(__dirname, '../../../suite-desktop');
     const desiredLogLevel = process.env.LOGLEVEL ?? 'error';
+    await TrezorUserEnvLink.api.startBridge();
```
Comment (mroz22) [inline packages/suite-desktop-core/e2e/support/common.ts:13]: well, this adding this line still means that node-bridge is not used in test

File: .github/workflows/test-suite-desktop-e2e.yml (hunk 6 above)
Comment (ondracja) [reply .github/workflows/test-suite-desktop-e2e.yml]: removed unnecessary `connect-iframe` build, the rest is still necessary after a discussion with @Nodonisko.

**Review by mroz22 — APPROVED** [review]
i don't understand why node-bridge is turned off. but ok. please fix

File: packages/suite-desktop-core/e2e/support/common.ts (hunk 8 above)
Comment (ondracja) [reply packages/suite-desktop-core/e2e/support/common.ts:13]: will figure out why node-bridge fails in an upcoming PR.

**Pull request merged**